src/main/scala/is/hail/distributedmatrix/HailBlockMatrix.scala

```diff
@@ -0,0 +1,704 @@
+package is.hail.distributedmatrix
+
+import java.io._
+
+import breeze.linalg.{DenseMatrix => BDM, _}
+import is.hail._
+import is.hail.utils._
+import org.apache.hadoop.io._
+import org.apache.spark._
+import org.apache.spark.mllib.linalg.distributed._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.storage.StorageLevel
+import org.json4s._
+
+object HailBlockMatrix {
```

Comment:
Can we really not call this `HailBlockMatrix`, but just `BlockMatrix`?  If we still need to import conflicting symbols from other packages (e.g. Spark), we should rename them on import (like we do with Vector and Matrix from Spark vs Breeze).  It is clear it is the Hail block matrix because it is in the `is.hail` package.